content/docs/state-and-lifecycle.md

```diff
@@ -300,84 +300,84 @@ ReactDOM.render(
+४) प्रत्येक सेकेन्ड ब्राउजरले `tick()` विधि कल गर्दछ। यसको भित्र `Clock` कम्पोनेंटले हालको समय समावेश गरेको objectसँग `setState()` कल गरेर UI अपडेट अनुसूची गर्छ। यो `setState()` कलको कारण Reactलाई थाहा छ state परिवर्तन भएको छ र स्क्रिनमा के हुनुपर्दछ भनेर जान्न उसले फेरि `render()` विधिलाई कल गर्दछ। यस पटक `this.state.date`को `render()` विधि फरक हुनेछ र त्यसैले रेन्डर आउटपुटमा अपडेट गरिएको समय समावेश हुनेछ। React तदनुसार DOM अपडेट गर्दछ।
 
-5) If the `Clock` component is ever removed from the DOM, React calls the `componentWillUnmount()` lifecycle method so the timer is stopped.
+५) यदि `Clock` कम्पोनेंट कहिले पनि DOM बाट हटाइन्छ, Reactले `componentWillUnmount()` लाइफसाइकल विधि कल गर्दछ त्यसैले टाइमर रोकिन्छ।
 
-## Using State Correctly {#using-state-correctly}
+## कसरी Stateको सही प्रयोग गर्ने {#using-state-correctly}
 
-There are three things you should know about `setState()`.
+त्यहाँ तीनवटा कुरा छन् जुन तपाईले `setState()`को बारेमा जान्नुपर्दछ।
 
-### Do Not Modify State Directly {#do-not-modify-state-directly}
+### Stateको सिधा परिमार्जन नगर्नुहोस् {#do-not-modify-state-directly}
 
-For example, this will not re-render a component:
+उदाहरण को लागी, यसले कम्पोनेंट पुन: रेंडर गर्दैन:
 
 ```js
-// Wrong
+// गलत
 this.state.comment = 'Hello';
 ```
 
-Instead, use `setState()`:
+यसको सट्टा `setState()` प्रयोग गर्नुहोस्
 
 ```js
-// Correct
+// सही
 this.setState({comment: 'Hello'});
 ```
 
-The only place where you can assign `this.state` is the constructor.
+कन्स्ट्रक्टर एक मात्र स्थान जहाँ तपाईं `this.state`मा मान प्रदान गर्न सक्नुहुन्छ।
 
-### State Updates May Be Asynchronous {#state-updates-may-be-asynchronous}
+### State अपडेटहरू एसिन्क्रोनस हुन सक्छ {#state-updates-may-be-asynchronous}
 
-React may batch multiple `setState()` calls into a single update for performance.
+Reactले कार्यसम्पादन क्षमताको लागि धेरै `setState()` कलहरूलाई एक अपडेटमा ब्याच गर्न सक्दछ।
 
-Because `this.props` and `this.state` may be updated asynchronously, you should not rely on their values for calculating the next state.
+किनकि `this.prop` र `this.state` asynchronously अपडेट हुन सक्छ, तपाईंले अर्को state गणना गर्न उनीहरूको मानहरूमा भर पर्नुहुन्न।
 
-For example, this code may fail to update the counter:
+उदाहरण को लागी, यो कोड काउन्टर अपडेट गर्न असफल हुन सक्छ
 
 ```js
-// Wrong
+// गलत
 this.setState({
   counter: this.state.counter + this.props.increment,
 });
 ```
 
-To fix it, use a second form of `setState()` that accepts a function rather than an object. That function will receive the previous state as the first argument, and the props at the time the update is applied as the second argument:
+यसलाई ठीक गर्नका लागि `setState()` को एक दोस्रो फारम प्रयोग गर्नुहोस् जुन एक वस्तुको सट्टा function स्वीकार गर्दछ। त्यो functionले पहिलो आर्गुमेन्टको रूपमा अघिल्लो state प्राप्त गर्दछ र अपडेट लागू भएको बेलाको propsहरू दोस्रो आर्गुमेन्टको रूपमा प्राप्त गर्दछ:
```

Comment:
`फारम` means form as in something you fill. The `form` here means shape or configuration.